workshops/image_editor/README.md

```diff
@@ -0,0 +1,451 @@
+
+Repl.it should have automatically linked the `style.css` file for you, but if you don't see your changes reflect when you click the green "Run" button at the top, try adding this line to the `<head>` in your `index.html` file:
+
+```html
+<link rel="stylesheet" href="style.css">
+```
+
+# JavaScript
+This is my favorite part of the workshop, because this part makes the website actually work.
+
+Navigate to the `script.js` file in your sidebar. Then, create these 3 variables:
+
+```javascript
+let myform = document.getElementById('imgurl');
+let targetimage = document.getElementById('targetimage');
+let inputrange = document.querySelectorAll('.slider');
+```
+- The first and second variable will hold html elements, we get those elements using the method ```document.getElementById('')```. So those variables will hold the form in section 1 and the image in section 2 respectively.
+- In the third variable, you'll find `document.querySelectorAll` returns a NodeList, a special type of list of elements, that will contain the amount of elements with  `class=slider` in the document. So the 3rd variable contains the value of all the range inputs in the section 3.
+
+Next, we want to trigger some JavaScript code to run when the form is submitted. Under the three variables, add the following code:
+
+```js
+myform.addEventListener('submit', function(e) {
+  let urlimage = document.getElementById('onlineurl');
+  let urlimageval = urlimage.value;
+  if(urlimageval.length){
+    targetimage.setAttribute('src',urlimageval);
+      urlimage.value = '';
+  }
+    
+  e.preventDefault();
+  console.log('Image displayed');
+});
+```
+Let's break this down:
+We are using the method ```addEventListener()``` to the first variable that we created, when the button in the form is submitted we call a function that:
+1. Creates a variable that holds the text input that we placed in section 1.
+2. Creates another variable that will get the data gotten from the text input, we do this using the **value** property.
+3. Creates an if statement that checks if the variable `urlimageval` is empty or not using the **length** property, this property returns `true` if the length of the string is greater than 0 or `false` if the length of the string is 0. When the property returns true, it takes the `img` element in section 2 and sets its `src` to the value of `urlimage`, then it resets the value of `urlimage`.
+4. Then we use the `preventDefault();` method to prevent the default way the browser handles `submit` when it happens, which includes refreshing the page.
+5. Finally, we use the function `console.log('');` to send a message to the console.
+
+Now let's test it:
+
+![Image displaying gif](https://cloud-jumlfocs6.vercel.app/0screen_recording_2020-11-08_at_11.26.14_pm.gif)
+
+Works perfectly!
+
+Now we'll need to add an event listener to each range input in section 3, in order to be able to edit the image:
+```js
+for(let i=0; i<=inputrange.length-1; i++) {
```

Comment:
i feel like better code here would be `for (let i = 0; i < inputrange.length; i++)`